ServiceBoxWeb/src/main/java/com/infoshareacademy/dto/client/UpdateClientDto.java

```diff
@@ -0,0 +1,73 @@
+package com.infoshareacademy.dto.client;
+
+import com.infoshareacademy.entity.client.Address;
+import com.infoshareacademy.entity.client.Client;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+
+/**
+ * A DTO for the {@link Client} entity
+ */
+@Getter
+@Setter
+@NoArgsConstructor
+public class UpdateClientDto {
+    @NotBlank(message = "To pole jest wymagane")
+    private String name;
+    private AddressDto address;
+    @Pattern(regexp = "(^$|\\d{10})", message = "Niepoprawny NIP")
+    private String nip;
+    @NotBlank(message = "To pole jest wymagane")
+    @Pattern(regexp = "(^$|\\d{9})", message = "Niepoprawny numer telefonu")
+    private String phoneNumber;
+    @NotBlank(message = "To pole jest wymagane")
+    @Email
+    private String email;
+    private boolean allowNotifications;
+    private long id;
+
+    /**
+     * A DTO for the {@link Address} entity
+     */
+    @Getter
+    @Setter
+    @NoArgsConstructor
+    public static class AddressDto {
+        private String street;
+        private String houseNumber;
+        private String flatNumber;
+        @Pattern(regexp = "(^$|\\d{2}-\\d{3})", message = "Podaj kod pocztowy w formacie xx-xxx")
+        private String zipCode;
+        private String city;
+    }
+
+    public Client toClient() {
+        Address clientAddress = null;
+
+        if (this.address != null) {
+            clientAddress = new Address(
+                    this.address.street,
+                    this.address.houseNumber,
+                    this.address.flatNumber,
+                    this.address.zipCode,
+                    this.address.city
+            );
+        }
+        Client client = new Client(
+                this.name,
+                clientAddress,
+                this.nip,
+                this.phoneNumber,
+                this.email,
+                this.allowNotifications
+        );
+        client.setId(this.id);
+        return client;
```

Comment:
mapowanie powinno znalezc sie w oddzielanej klasie np. ClientMapper

Reply:
To całe DTO to chyba zapomnieliśmy usunąć?! Bo na ten moment nigdzie z tego nie korzystamy ponieważ przy update wykorzystujemy ClientDto.